apps/token-manager/app/src/screens/Holders.js

```diff
@@ -1,166 +1,163 @@
-                  >
-                    {holders.map(({ address, balance }) => (
-                      <HolderRow
-                        key={address}
-                        address={address}
-                        balance={balance}
-                        groupMode={groupMode}
-                        isCurrentUser={Boolean(
-                          userAccount && userAccount === address
-                        )}
-                        maxAccountTokens={maxAccountTokens}
-                        tokenDecimalsBase={tokenDecimalsBase}
-                        onAssignTokens={onAssignTokens}
-                        onRemoveTokens={onRemoveTokens}
-                        compact={compactTable}
-                      />
-                    ))}
-                  </ResponsiveTable>
-                )}
-              </Main>
-              {(!tabbedNavigation || selectedTab === 1) && (
-                <SideBar
-                  holders={holders}
-                  tokenAddress={tokenAddress}
-                  tokenDecimalsBase={tokenDecimalsBase}
-                  tokenName={tokenName}
-                  tokenSupply={tokenSupply}
-                  tokenSymbol={tokenSymbol}
-                  tokenTransfersEnabled={tokenTransfersEnabled}
-                  userAccount={userAccount}
-                />
-              )}
-            </TwoPanels>
-          )
-        }}
-      </Viewport>
-    )
-  }
+            return values
+          }}
+          renderEntryActions={([address, balance]) => (
+            <EntryActions
+              address={address}
+              balance={balance}
+              onAssignTokens={onAssignTokens}
+              onRemoveTokens={onRemoveTokens}
+              singleToken={balance.eq(tokenDecimalsBase)}
+              canAssign={balance.lt(maxAccountTokens)}
+            />
+          )}
+        />
+      }
+      secondary={
+        <InfoBoxes
+          holders={holders}
+          tokenAddress={tokenAddress}
+          tokenDecimalsBase={tokenDecimalsBase}
+          tokenName={tokenName}
+          tokenSupply={tokenSupply}
+          tokenSymbol={tokenSymbol}
+          tokenTransfersEnabled={tokenTransfersEnabled}
+          userAccount={userAccount}
+        />
+      }
+    />
+  )
+}
 
-  handleSelectTab = index => {
-    this.setState({ selectedTab: index })
-  }
+Holders.propTypes = {
+  holders: PropTypes.array,
+}
+Holders.defaultProps = {
+  holders: [],
 }
 
-const TabBarWrapper = styled.div`
-  margin-top: 16px;
-  & ul {
-    border-bottom: none !important;
-  }
-  & li {
-    padding: 0 20px;
-  }
-`
+function EntryActions({
+  address,
+  onAssignTokens,
+  onRemoveTokens,
+  singleToken,
+  canAssign,
+}) {
+  const theme = useTheme()
+  const [label, showLocalIdentityModal] = useIdentity(address)
 
-const ResponsiveTable = styled(Table)`
-  margin-top: 16px;
+  const editLabel = useCallback(() => showLocalIdentityModal(address), [
```

Comment:
Nitpick for consistency: all these handlers, could you update the names to `handleActionsToDo`?

Reply:
If it can add something to the [other discussion](https://github.com/aragon/aragon-apps/pull/918#discussion_r305695753):

I don’t consider these to be handle any event, even if they happen to be used as such later on. When a function is meant to handle an event specifically, I usually name it using the `handle` prefix with the emitted event − not the action it triggers.

For example here, a `handle`-prefixed name I could use would be `handleClick()`, since they “handle a click” (which would be a problem since they are three :laughing:). But they are not really specific to the click event, so it could be limiting to do that (e.g. if we want to call it from somewhere else at some point).

`handleEditLabel` wouldn’t work following the same logic, because the function doesn’t “handle an edit”, but kind of the opposite: it triggers an edit.

Reply:
`handleClickForEdit`?

Reply:
I don’t think we should limit it to handling clicks :laughing: I think it is more important to name this function after what it does (“edit a label”), rather than its usage (being passed to `onClick`). There is nothing related to handling a click event in this function, it might be called from anywhere.

Reply:
> it might be called from anywhere

This is what I'm pointing at, it might, but it is not, it is being used in a click handler. When the methods are specific to an implementation it doesn't need a generic name, if/when it responds to some other type of event maybe the abstract name should be updated but for now, only one use case: click.